### PR TITLE
Make deprecated objects not error out with --allow-deprecated

### LINKED ISSUE #7304: Errors get generated by deprecated objects with --allow-deprecated and --error
### Description of the enhancement or error report
If you run a MOOSE app with `--error` and `--allow-deprecated`, the deprecated code objects or parameters generate warnings, which get turned to errors, and cause a failure.  Because this is the way the test harness runs the code, you can't currently test deprecated objects even with the `allow_deprecated_until` parameter in the tests file without generating an error. Running with `--allow-deprecated` should always generate a warning if we're before the deprecation date.

### Rationale for the enhancement or information for reproducing the error
See above.

### Identified impact
The code that handles these errors currently calls  `mooseWarning()`.  Since those get translated to errors if we're running with `--error`, you always get an error.  I believe the correct fix for this is to simply call `mooseDeprecated()` instead, which correctly generates a warning or error.

### PR DESCRIPTION
**Pull request by bwspenc**
closes #7304

### CODE REVIEW

File: framework/src/base/Factory.C [hunk 1]
```diff
@@ -154,7 +154,7 @@ void Factory::deprecatedMessage(const std::string obj_name)
       msg << "Replaced " << obj_name << " with " <<  name_it->second;
 
     // Produce the error message
-    mooseDoOnce(mooseWarning(msg.str()));
+    mooseDoOnce(mooseDeprecated(msg.str()));
```
Comment (permcody) [inline framework/src/base/Factory.C]: You don't need the `mooseDoOnce()` anymore here since that macro is inside of `mooseDeprecated()`

**Comment by permcody** [comment]
Does this actually fix your issue? It looks like you should still have the same problem because `mooseDeprecated()` calls `mooseError()` if the global flag is set.

**Comment by bwspenc** [comment]
@permcody Yes, this does fix the issue. In `MooseApp.C` `_deprecated_is_error` doesn't get set to true if `--allow-deprecated` is used.

**Comment by permcody** [comment]
Then how are you getting passed the check inside of `mooseDeprecated`?
https://github.com/idaholab/moose/blob/devel/framework/include/base/MooseError.h#L149-L152

**Comment by bwspenc** [comment]
_deprecated_is_error is false in this case.

**Comment by permcody** [comment]
OK - I see now what's going on. Thanks

**Pull request merged**